elixir/apps/domain/lib/domain/auth/adapters/microsoft_entra/jobs.ex

```diff
@@ -68,7 +68,7 @@ defmodule Domain.Auth.Adapters.MicrosoftEntra.Jobs do
       |> Ecto.Multi.update(:save_last_updated_at, fn _effects_so_far ->
         Auth.Provider.Changeset.sync_finished(provider)
       end)
-      |> Domain.Repo.transaction()
+      |> Domain.Repo.transaction(timeout: :timer.minutes(15))
```

Comment:
Just wondering if it's possible for a second provider sync to start within this time frame?  Just wondering what might happen if this transaction was hung up for 15 min and then another transaction tries to start for the same provider?

Reply:
We have a pool for Ecto so we should be fine for now and #3801 will fix it long term

Reply:
We should also use `Stream` later to limit concurrency

Reply:
👍